exporter/zipkin/lib/opentelemetry/exporter/zipkin/transformer.rb

```diff
@@ -0,0 +1,180 @@
+        KIND_MAP = {
+          OpenTelemetry::Trace::SpanKind::SERVER => 'SERVER',
+          OpenTelemetry::Trace::SpanKind::CLIENT => 'CLIENT',
+          OpenTelemetry::Trace::SpanKind::PRODUCER => 'PRODUCER',
+          OpenTelemetry::Trace::SpanKind::CONSUMER => 'CONSUMER',
+          OpenTelemetry::Trace::SpanKind::INTERNAL => nil
+        }.freeze
+
+        SERVICE_NAME_ATTRIBUTE_KEY = 'service.name'
+        ERROR_TAG_KEY = 'error'
+        STATUS_CODE_NAME = 'otel.status_code'
+        STATUS_UNSET = 'UNSET'
+        STATUS_ERROR = 'ERROR'
+        STATUS_OK = 'OK'
+        ATTRIBUTE_PEER_SERVICE = 'peer.service'
+        ATTRIBUTE_NET_PEER_IP = 'peer.ip'
+        ATTRIBUTE_NET_PEER_PORT = 'peer.port'
+        ATTRIBUTE_NET_HOST_IP = 'host.ip'
+        ATTRIBUTE_NET_HOST_PORT = 'host.port'
+
+        DEFAULT_SERVICE_NAME = OpenTelemetry::SDK::Resources::Resource.default.attribute_enumerator.find { |k, _| k == SERVICE_NAME_ATTRIBUTE_KEY }&.last || 'unknown_service'
+        private_constant(:KIND_MAP, :DEFAULT_SERVICE_NAME, :SERVICE_NAME_ATTRIBUTE_KEY, :ERROR_TAG_KEY, :STATUS_CODE_NAME, :STATUS_UNSET, :STATUS_ERROR, :STATUS_OK)
+
+        def to_zipkin_span(span_d, resource)
+          start_time = (span_d.start_timestamp.to_f * 1_000_000).to_i
+          duration = (span_d.end_timestamp.to_f * 1_000_000).to_i - start_time
+          tags = {}
+          service_name = DEFAULT_SERVICE_NAME
+          resource&.attribute_enumerator&.select do |key, value|
+            service_name = value if key == SERVICE_NAME_ATTRIBUTE_KEY
+          end
+
+          add_il_tags(span_d, tags)
+          add_status_tags(span_d, tags)
+          tags = aggregate_span_tags(span_d, tags)
+
+          # TOOO: set debug flag? (is that represented in tracestate?)
+          # https://github.com/openzipkin/b3-propagation#why-is-debug-encoded-as-x-b3-flags-1
+          # https://github.com/openzipkin/zipkin-api/blob/7692ca7be4dc3be9225db550d60c4d30e6e9ec59/zipkin2-api.yaml#L475
+          # TODO: shared key mapping
+
+          zipkin_span = {
+            name: span_d.name,
+            traceId: span_d.hex_trace_id,
+            id: span_d.hex_span_id,
+            timestamp: start_time,
+            duration: duration,
+            debug: false
+          }
+
+          add_conditional_tags(zipkin_span, span_d, tags, service_name)
+          add_annotations(zipkin_span, span_d)
+
+          zipkin_span
+        end
+
+        def add_il_tags(span_data, tags)
+          tags['otel.library.name'] = span_data.instrumentation_library&.name if span_data.instrumentation_library&.name
+          tags['otel.library.version'] = span_data.instrumentation_library&.version if span_data.instrumentation_library&.version
```

Comment:
```suggestion
          tags['otel.library.name'] = span_data.instrumentation_library.name
          tags['otel.library.version'] = span_data.instrumentation_library.version
```
`span_data.instrumentation_library` should always be non-nil.